src/libs/actions/Report.js

```diff
@@ -204,13 +204,14 @@ function fetchChatReportsByIDs(chatList) {
  *
  * @param {Number} reportID
  * @param {Number} sequenceNumber
+ * @param {Number} maxSequenceNumber
  */
-function setLocalLastRead(reportID, sequenceNumber) {
+function setLocalLastRead(reportID, sequenceNumber, maxSequenceNumber = 0) {
     lastReadSequenceNumbers[reportID] = sequenceNumber;
 
     // Update the report optimistically
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
+        unreadActionCount: maxSequenceNumber > 0 ? maxSequenceNumber - sequenceNumber : 0,
```

Comment:
- I'm not sure about why we need to pass the `maxSequenceNumber` in from the outside here. I think we can maybe see what the `maxSequenceNumber` is by looking somewhere in `actions/Report`?

- Using `Math.max()` might clean up the conditional.

- Also, slightly unsure about when the `sequenceNumber` would be different from the `maxSequenceNumber` probably it would be good to add a comment that explains this.

Reply:
SequenceNumber will be different, for example, in this flow, when we mark an older sequence as unread

Reply:
Ok it looks better to me now and I think is self explanatory looking again. Thanks :)